caf-audit-binding-elasticsearch/src/main/java/com/hpe/caf/auditing/elastic/ElasticAuditConnection.java

```diff
@@ -32,7 +35,33 @@ public ElasticAuditConnection() throws AuditConfigurationException
             final String hostAndPorts = 
                 System.getProperty(ElasticAuditConstants.ConfigEnvVar.CAF_ELASTIC_HOST_AND_PORT_VALUES,
                                    System.getenv(ElasticAuditConstants.ConfigEnvVar.CAF_ELASTIC_HOST_AND_PORT_VALUES));
-
+            final String hostValues = 
+                System.getProperty(ElasticAuditConstants.ConfigEnvVar.CAF_ELASTIC_HOST_VALUES,
+                                   System.getenv(ElasticAuditConstants.ConfigEnvVar.CAF_ELASTIC_HOST_VALUES));
+            final String port = 
+                System.getProperty(ElasticAuditConstants.ConfigEnvVar.CAF_ELASTIC_PORT_VALUE,
+                                   System.getenv(ElasticAuditConstants.ConfigEnvVar.CAF_ELASTIC_PORT_VALUE));
+            final StringBuilder hostAndPortBuilder = new StringBuilder();
+            final String[] hostArray;
+            
+            if(hostAndPorts == null){
+                hostArray = hostValues.split(",");
+                if (hostArray.length == 0) {
+                    final String errorMessage = "No hosts configured.";
+                    LOG.error(errorMessage);
+                    throw new AuditConfigurationException(errorMessage);
+                }
+                for (int index = 0; index < hostArray.length; index++){
+                    hostAndPortBuilder.append(hostArray[index]).append(':') .append(port).append(',');
+                }
+            }
+            
+            String hostAndPortsStr = "";
```

Comment:
What is the point of assigning an empty string when you're about to overwrite it?  And why aren't you using `final`?  This should be:

    final String hostAndPortsStr;